src/include/btree_interval_set.h

```diff
@@ -0,0 +1,588 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#ifndef CEPH_BTREE_INTERVAL_SET_H
+#define CEPH_BTREE_INTERVAL_SET_H
+
+#include <iterator>
+#include <map>
+#include <ostream>
+using namespace std;
```

Comment:
`using namespace` in header files is not a good practice in general, as it brings the entire namespace into the global namespace. as a user, I wouldn't expect that behavior from `#include "btree_interval_set.h"`. the std namespace has tons of names, so conflicts are likely in a large codebase, and they're hard to track down because they generally show up in templates or function overload resolution

all of the std types appear to be qualified with `std::` already, so you can probably just take that line out